### PR TITLE
DOCSP-9546, 9606: landing page and compatibility page

### PR DESCRIPTION
**Pull request by ccho-mongodb**
JIRA:
https://jira.mongodb.org/browse/DOCSP-9546
https://jira.mongodb.org/browse/DOCSP-9606

Staging: 
https://docs-mongodbcom-staging.corp.mongodb.com/3d846c0/java/docsworker-xlarge/DOCSP-9546-landing-page/

### CODE REVIEW

File: source/index.txt [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+   /compatibility
+
+Introduction
+------------
+
+The official MongoDB Java Driver enables Java applications to connect to
+MongoDB and work with data.
+
+If you are working with asynchronous stream processing and reactive
+streams, we recommend the Reactive Streams Driver.
```
Comment (terakilobyte) [inline source/index.txt]: Should we include a blank link here so it's not forgotten that this should link to something?

File: source/index.txt [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+Usage Examples
+--------------
+The :doc:`Usage Examples </usage-examples>` section provides
+runnable code snippets and explanations for common methods. We recommend
+this section for users that are new to the MongoDB Java driver.
+
+Fundamentals
+------------
+Whether you are new to MongoDB or you just want to brush up on the core
+concepts you can check out the **Fundamentals** section which contains
```
Comment (terakilobyte) [inline source/index.txt]:
```suggestion
concepts, you can check out the **Fundamentals** section which contains
```

File: source/includes/mongodb-compatibility-table-java.rst [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+     -
+     - ✓
+     - ✓
+
+   * - Version 2.14
+     -
+     -
+     -
+     -
+     - ✓ [*]_
```
Comment (terakilobyte) [inline source/includes/mongodb-compatibility-table-java.rst]: This results in a clickable link being inserted. Would `(*)` work?

Comment (ccho-mongodb) [reply source/includes/mongodb-compatibility-table-java.rst]: This is meant to be a footnote, but it is not working yet in nextgen. I created a ticket this morning: https://jira.mongodb.org/browse/DOP-931

File: source/index.txt (hunk 1 above)
Comment (ccho-mongodb) [reply source/index.txt]: I guess I'll just link to the current Reactive Streams driver reference docs since we're migrating one at a time.

**Comment by ccho-mongodb** [comment]
Ready for external review @jyemin

File: source/includes/language-compatibility-table-java.rst [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+     - ✓
+
+   * - Versions >= 2.7
+     - ✓
+     - ✓
+     - ✓
+     - ✓
+     - ✓
+
+.. [*] Java versions 8 and above are all supported thanks to the JVM backwards compatibility promise. Only LTS versions will be explicitly listed in future.
```
Comment (jyemin) [inline source/includes/language-compatibility-table-java.rst:90]: There is no JVM backward compatibility promise per se.  We are starting to see deprecated APIs being removed in newer JVM versions.

Comment (jyemin) [inline source/includes/language-compatibility-table-java.rst:90]: say "LTS Java versions", as we may introduce an LTS notion for MongoDB in the future, so the term may become ambiguous.

File: source/includes/mongodb-compatibility-table-java.rst [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+   * - Version 2.12
+     -
+     -
+     -
+     -
+     -
+     -
+     - ✓
+
+.. [*] The 2.14 driver does not support all MongoDB 3.2 features (e.g. read concern); however, if you are currently on a version 2.x driver and would like to run against MongoDB 3.2 but cannot upgrade to driver version 3.2, use the 2.14 driver.
```
Comment (jyemin) [inline source/includes/mongodb-compatibility-table-java.rst:159]: I think it's time to stop listing 2.x drivers here.

Comment (ccho-mongodb) [reply source/includes/mongodb-compatibility-table-java.rst:159]: Should 2.x versions of the driver also be removed from the Language Compatibility table?

Comment (jyemin) [reply source/includes/mongodb-compatibility-table-java.rst]: Remove 2.x everywhere IMO

**Pull request merged**